ChaosMod/Effects/db/Player/PlayerTpToRandomStore.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+    Effect by MoneyWasted
+*/
+
+#include <stdafx.h>
+
+static void RandomStore(int random)
+{
+    switch (random)
+    {
+    case 0:
+        TeleportPlayer(372.29217529297f, 326.39370727539f, 103.56636047363f);   // 247 Clinton
+        break;
```

Comment:
Thinking about it, this might be able to be cleaned up with an array of vectors?

```cpp
Vector3 STORES [] = {
  {372.29217529297f, 326.39370727539f, 103.56636047363f}, // 247 Clinton
  ...
};

...

static void TPVector(Vector3 coords) {
  TeleportPlayer(coords.x, coords.y, coords.z);
}

static void RandomStore(int random) {
  TPVector(STORES[random]);
}
```

Reply:
Geez how did I not think of that. Seems like a much cleaner approach. I’ll get that implemented.

Reply:
Frankly, with this approach you can probably do away with the `RandomStore` method entirely and just fire `TPVector`  in the onStart

Reply:
Yea I could just make one function that handles the teleporting and the random selection for simplicity sake.

Reply:
Updated to use new method